# Pangenome Graphs in Infectious Disease: A Promising and Practical Implementation for Comprehensive Genetic Variation Analysis 

Zuyu Yang1, Andrea Guarracino2,7, Xiaoyun Ren1, Patrick J. Biggs3,4, Nuzla Ismail5, Michael Black5, Tony R Merriman5,6, Pjotr Prins7, Erik Garrison7, Joep de Ligt1*
1. Institute of Environmental Science and Research, Porirua, New Zealand 
2. Genomics Research Centre, Human Technopole, Milan, Italy 
3. Molecular Biosciences Group, School of Natural Sciences, Massey University, Palmerston North, New Zealand
4. Molecular Epidemiology & Public Health Laboratory, School of Veterinary Science, Massey University, Palmerston North, New Zealand
5. Department of Biochemistry, University of Otago, New Zealand
6. Division of Clinical Immunology and Rheumatology, University of Alabama at Birmingham, Alabama, USA
7. Department of Genetics, Genomics and Informatics, University of Tennessee Health Science Center, Tennessee, USA
* Correspondening author 

**Keywords:** Pangenome graphs, Infectious diseases, Genomic surveillance, Comparative genomics, Genetic variation, Long-read sequencing, Genome assembly

## Abstract
Whole genome sequencing has revolutionized infectious diseases surveillance for tracking and monitoring the spread and evolution of pathogens. 
However, using linear reference-based approach for genomic analyses may result in biases, especially when studies are conducted on highly variable bacterial genomes. 
Pangenome graphs provide an efficient model for representing and analyzing multiple genomes and their variants as a graph structure that includes all types of variations.
In this study, we present a practical bioinformatic pipeline that employs pggb and vg toolkits to build pangenomes from assembled genomes, align whole genome sequencing data and call variants against a graph reference. 
The pangenome graph enables the identification of structural variants, rearrangements, and small variants (e.g., single nucleotide polymorphisms and insertions/deletions) simultaneously. 
We demonstrate that using a pangenome graph, instead of a single linear reference genome, improves mapping rates and variant calling for both simulated and real datasets of Neisseria meningitidis. 
Overall, pangenome graphs offer a promising approach for comparative genomics and comprehensive genetic variation analysis in infectious diseases. 

## Introduction


