\begin{abstract}

%%% Leave the Abstract empty if your article does not require one, please see the Summary Table for full details.
\section{}
For full guidelines regarding your manuscript please refer to \href{http://www.frontiersin.org/about/AuthorGuidelines}{Author Guidelines}.

Whole genome sequencing has revolutionized infectious diseases surveillance for tracking and monitoring the spread and evolution of pathogens. However, using linear reference-based approach for genomic analyses may result in biases, especially when studies are conducted on highly variable bacterial genomes. Pangenome graphs provide an efficient model for representing and analyzing multiple genomes and their variants as a graph structure that includes all types of variations. In this study, we present a practical bioinformatic pipeline that employs pggb and vg toolkits to build pangenomes from assembled genomes, align whole genome sequencing data and call variants against a graph reference. The pangenome graph enables the identification of structural variants, rearrangements, and small variants (e.g., single nucleotide polymorphisms and insertions/deletions) simultaneously. We demonstrate that using a pangenome graph, instead of a single linear reference genome, improves mapping rates and variant calling for both simulated and real datasets of Neisseria meningitidis. Overall, pangenome graphs offer a promising approach for comparative genomics and comprehensive genetic variation analysis in infectious diseases. 


\tiny
 \keyFont{ \section{Keywords:} Pangenome graphs, Infectious diseases, Genomic surveillance, Comparative genomics, Genetic variation, Long-read sequencing, Genome assembly} %All article types: you may provide up to 8 keywords; at least 5 are mandatory.
\end{abstract}
