For outbreak detection for infectious diseases genomics has proven to be a valuable datasource \citep{article}. Most of these investigations use short read data \citep{article} and utilise a reference genome for downstream mapping and variant identification, often focussed on the core genome \citep{article}. A major caveat of this approach is the impact the chosen reference genome has on the resolution of the phylogenetic tree and therefore the ability to reliably detect genetic clusters and therefore outbreaks \citep{article}. Pangenome graphs promise to be a solution that can tackle several of the shortcomings through a data structure with less to no reference bias as well as a broader spectrum of variants being readily detectable and a more dynamic definition of core genomes. Here we explore the performance and utility of these tools in a public health setting.

Over the past few decades, whole genome sequencing (WGS) has become an indispensable tool in the field of infectious disease research, surveillance, and control[1, 2] . Rapid advancements in sequencing technologies and bioinformatics analysis have facilitated the generation of high-quality genomic data at an unprecedented scale[3, 4]. WGS has enabled researchers to track and monitor the spread and evolution of pathogens, investigate outbreaks, identify drug resistance markers, and develop diagnostic assays and vaccines[5-9].  Its utility has been especially evident in the SARs-CoV-2 pandemic such as  real-time tracking of the pandemic[10] and identifying transmission chains[11]. Additionally, WGS has provided valuable insights into the genetic diversity, population structure, and functional characteristics of various pathogens, thereby shaping our understanding of the molecular mechanisms driving their virulence and transmission[12].
Currently, genomic surveillance concentrates on monitoring lineages and establishing linkages between cases. Analysis is mainly dependent on mutations in the core genome, the genomic regions that are common to all isolates, using one linear genome as a reference. Bacteria genomes are highly variable, genomic rearrangements and different-scale deletion or insertion events are common[13]. Using a single reference approach, variations in the accessory genome (regions not in common) are not detected, suggesting we may miss important variations and  introduce biases due to reference selection. Consequently, the alignment of sequencing data against a single reference genome may lead to inaccurate or incomplete variant identification (Garrison & Marth, 2012, freebayes). Moreover, the linear representation of a genome fails to capture the complexity of genomic rearrangements, duplications, and structural variants that are critical for understanding pathogen evolution and adaptation[14, 15]. 
To overcome these limitations, pangenome graphs have emerged as an alternative approach for representing and analyzing multiple genomes and their variants[16, 17]. A pangenome graph is a graph-based data structure that captures the entire genomic diversity of a set of related organisms by incorporating all types of variations, including structural variants, rearrangements, and small variants (e.g., single nucleotide polymorphisms and insertions/deletions)[18, 19]. By representing genomes as graphs, pangenome graphs allow for more accurate and comprehensive genetic variation analysis, as they provide a unified framework to compare and analyze diverse genomes, overcoming the biases associated with single linear reference genomes[19].
The implementation of pangenome graphs in infectious disease research is important and advantageous First, the use of pangenome graphs allows for the identification of novel genetic variants and structural variations that may be missed by traditional linear reference-based methods[16]. Pangenome graphs enable improved mapping rates and variant calling, as demonstrated by the Neisseria meningitidis datasets in this study. This enhanced accuracy by including all type of variants can improve outbreak investigations, drug resistance prediction, and vaccine design[20, 21]. Pangenome graphs offer a promising and practical approach for comparative genomics and comprehensive genetic variation analysis in infectious diseases, paving the way for more accurate and in-depth investigations of pathogen diversity, evolution, and adaptation[17, 19].

