For outbreak detection for infectious diseases genomics has proven to be a valuable datasource \citep{article}. Most of these investigations use short read data \citep{article} and utilise a reference genome for downstream mapping and variant identification, often focussed on the core genome \citep{article}. A major caveat of this approach is the impact the chosen reference genome has on the resolution of the phylogenetic tree and therefore the ability to reliably detect genetic clusters and therefore outbreaks \citep{article}. Pangenome graphs promise to be a solution that can tackle several of the shortcomings through a data structure with less to no reference bias as well as a broader spectrum of variants being readily detectable and a more dynamic definition of core genomes. Here we explore the performance and utility of these tools in a public health setting.